From 7fe322b9cedae313cd9af6f403eab9bfc6027674 Mon Sep 17 00:00:00 2001
From: Jan Tojnar <jtojnar@gmail.com>
Date: Mon, 28 Mar 2022 21:02:23 +0200
Subject: [PATCH] online-accounts: Fix goa-helper on X11
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The handle_str contains the “x11:” prefix so strtol would return 0.
We need to pass it the numeric substring.

Fixes: https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/1651
---
 panels/online-accounts/gnome-control-center-goa-helper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/online-accounts/gnome-control-center-goa-helper.c b/panels/online-accounts/gnome-control-center-goa-helper.c
index 56513147e..533609cbc 100644
--- a/panels/online-accounts/gnome-control-center-goa-helper.c
+++ b/panels/online-accounts/gnome-control-center-goa-helper.c
@@ -129,7 +129,7 @@ set_external_parent_from_handle (GtkApplication *application,
           int xid;
 
           errno = 0;
-          xid = strtol (handle_str, NULL, 16);
+          xid = strtol (handle_str + strlen (x11_prefix), NULL, 16);
           if (errno != 0)
             {
               g_warning ("Failed to reference external X11 window, invalid XID %s", handle_str);
-- 
2.35.1

